server/routes/profile.go

```diff
@@ -94,8 +101,98 @@ func GetProfile(w http.ResponseWriter, r *http.Request) {
 }
 
 func EditProfile(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPatch {
+		log.Printf("Invalid request method")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Write([]byte("Invalid request method"))
+		return
+	}
+
+	//Connect to datastore
+	ctx := r.Context()
+	dsClient, err := datastore.NewClient(ctx, config.Project())
+	if err != nil {
+		log.Printf("Cannot connect to DataStore: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("datastore error"))
+		return
+	}
+
+	//Retrieve user from datastore
+	vars := mux.Vars(r)
+	userID := vars["id"]
+	if len(userID) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("no ID provided"))
+		return
+	}
+
+	k := datastore.NameKey("User", userID, nil)
+
+	var user models.User
+	if err := dsClient.Get(ctx, k, &user); err != nil {
+		log.Printf("Cannot retrieve user from DataStore: %v", err)
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte("entity not found"))
+		return
+	}
+
+	//Get request body
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Cannot read request body: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("error reading body"))
+		return
+	}
+
+	//Get patch operations from body
+	var bodyMapped []struct {
+		Op    string
+		Path  string
+		Value string
+	}
+	err = json.Unmarshal(body, &bodyMapped)
+	if err != nil {
+		log.Printf("Cannot read patch operations: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("error reading patch operations"))
+		return
+	}
+
+	//Read through operations and apply valid changes
+	for _, v := range bodyMapped {
+		if v.Op == "replace" {
+			switch v.Path {
+			case "/Name":
+				user.Name = v.Value
+			case "/Bio":
+				user.Bio = v.Value
+			case "/ProfilePic":
+				user.ProfilePic = v.Value
+			}
+		}
+	}
+
+	//Save patched user
+	k, err = dsClient.Put(ctx, k, &user)
+	if err != nil {
+		log.Printf("Cannot save user to DataStore: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("error storing new user"))
+		return
+	}
+
+	//Return updated user
+	out, err := json.MarshalIndent(user, "", "  ")
```

Comment:
I feel like you don't need to add indentation here, since you're just sending additional bytes over the network